### PR TITLE
assets: move deposit components from react-invenio-deposit

### PR DESCRIPTION
**Pull request by kpsherva**
* closes https://github.com/inveniosoftware/invenio-rdm-records/issues/1276

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -83,3 +90,15 @@ jobs:
       - name: Run tests
         run: |
           ./run-tests.sh
+
+      - name: Install deps for frontend tests
+        working-directory: ./invenio_administration/assets/semantic-ui/js/invenio_administration
```
Comment (alejandromumo) [inline .github/workflows/tests.yml:95]: Is this directory correct (`invenio_administration`)?

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/index.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// This file is part of InvenioAdministration
```
Comment (alejandromumo) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/index.js:1]:
```suggestion
// This file is part of Invenio-RDM-Records
```

There are other occurrences of this in the PR

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/customFields/CustomField.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    deserializedDefault = null,
+    serializedDefault = null,
+    allowEmpty = false,
+    vocabularyFields = [],
+  }) {
+    super({ fieldpath, deserializedDefault, serializedDefault, allowEmpty });
+    this.vocabularyFields = vocabularyFields;
+  }
+
+  #mapCustomFields(record, customFields, mapValue) {
```
Comment (alejandromumo) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/customFields/CustomField.js:26]: Not added in this PR but the name starting with `#` triggered my curiosity.

**Review by alejandromumo — COMMENTED** [review]
LGTM.

minor: there are multiple headers outdated (e.g. still mentioning `react-invenio-deposit`). I also realised we do not have a lot of documentation (jsdocs) on many react components, however it is just a thought (it's completely out of scope from this issue).

major: tests are failing since the `MANIFEST.in` needs to be updated. 
major: gh actions is complaining, I suppose it's due to formatting / linting issues. But let's see if tests pass after the manifest is updated.

Note: I did not review every component's code, as that was already done in `react-invenio-deposit` (and we're juts moving code). I checked however that the file structure is the correct one to be picked by the alias `@js/invenio_rdm_records` and other minor changes to `invenio-rdm-records` itself (e.g. gh workflows)

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/customFields/CustomField.js (hunk 3 above)
Comment (kpsherva) [reply invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/customFields/CustomField.js:26]: its a private class attr (if that is what you are asking about): https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -52,6 +52,13 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v2
 
+      - name: Use Node.js ${{ matrix.node-version }}
```
Comment (ntarocco) [inline .github/workflows/tests.yml:55]: Same here missing `node-version`

**Review by ntarocco — APPROVED** [review]

**Pull request merged**